example/static_pivoting/examples_helper.hpp

```diff
@@ -0,0 +1,155 @@
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Siva Rajamanickam (srajama@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+
+template <class XType>
+void write2DArrayToMM(std::string name, const XType x) {
+  std::ofstream myfile;
+  myfile.open(name);
+
+  auto x_h = Kokkos::create_mirror_view(x);
+
+  Kokkos::deep_copy(x_h, x);
+
+  if (XType::Rank == 2) {
+    myfile << "%% MatrixMarket 2D Array\n%" << std::endl;
+    myfile << x_h.extent(0) << " " << x_h.extent(1) << std::endl;
+
+    for (size_t i = 0; i < x_h.extent(0); ++i) {
+      for (size_t j = 0; j < x_h.extent(1); ++j) {
+        myfile << std::setprecision(15) << x_h(i, j) << " ";
+      }
+      myfile << std::endl;
+    }
+
+    myfile.close();
+  }
+}
+
+template <class XType>
+void write3DArrayToMM(std::string name, const XType x) {
+  std::ofstream myfile;
+  myfile.open(name);
+
+  auto x_h = Kokkos::create_mirror_view(x);
+
+  Kokkos::deep_copy(x_h, x);
+
+  if (XType::Rank == 3) {
+    myfile << "%% MatrixMarket 3D Array\n%" << std::endl;
+    myfile << x_h.extent(0) << " " << x_h.extent(1) << " " << x_h.extent(2)
+           << std::endl;
+
+    for (size_t i = 0; i < x_h.extent(0); ++i) {
+      myfile << "Slice " << i << std::endl;
+      for (size_t j = 0; j < x_h.extent(1); ++j) {
+        for (size_t k = 0; k < x_h.extent(2); ++k) {
+          myfile << std::setprecision(15) << x_h(i, j, k) << " ";
+        }
+        myfile << std::endl;
+      }
+    }
+
+    myfile.close();
+  }
+}
+
+template <typename MatrixViewType, typename VectorViewType>
+void create_saddle_point_matrices(const MatrixViewType &A,
```

Comment:
You might want to add a little comment to explain what n_2 is?
And maybe just explain the structure of the saddle point problem for future users?